addons/udes_delivery_control/models/picking_extra_fields.py

```diff
@@ -0,0 +1,129 @@
+from odoo import api, fields, models, _
+from ..misc import date_diff, validate_dates
+
+
+class StockPickingExtras(models.Model):
+    """
+        Model to hold all extra fields of stock.picking object
+    """
+
+    _name = "stock.picking.extras"
+    _description = "UDES Picking Extra Fields"
+
+    picking_ids = fields.One2many(
+        comodel_name="stock.picking", inverse_name="u_extras_id", string="Pickings"
+    )
+    # Unloading Fields
+    user_id = fields.Many2one("res.users", string="User", help="Picking completed by user")
+    location_id = fields.Many2one("stock.location", domain=[("usage", "=", "internal")])
```

Comment:
Just check with product, but I'm not sure we need the location field as we only need to capture the lane as free text.

Reply:
I checked with Craig and he said if we are making this module generic (which we are) than probably it is best to keep it and make it optional field